src/mlpack/methods/ann/layer/inception_v3_impl.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file inception_v3_impl.hpp
+ * @author Shah Anwaar Khalid
+ *
+ * Implementation of the InceptionV3 Layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it uder the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_INCEPTION_V3_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_INCEPTION_V3_IMPL_HPP
+
+// In case it is not included.
+#include "inception_v3.hpp"
+
+namespace mlpack {
+namespace ann { /** Artificial Neural Network. */
+
+template<typename InputType, typename OutputType, int module>
+Inception3<InputType, OutputType, module> :: Inception3(
+    const size_t inSize,
+    const size_t inputWidth,
+    const size_t inputHeight,
+    const size_t outA,
+    const size_t outBOne,
```

Comment:
I would use more descriptive parameter names or maybe use a vector (`arma::vec`) as input to keep the number of parameters down.